pallets/account-manager/src/manager.rs

```diff
@@ -0,0 +1,263 @@
+
+impl<T: Config> AccountManagerExt<T::AccountId, BalanceOf<T>> for Pallet<T> {
+    fn deposit(
+        payee: &T::AccountId,
+        recipient: &T::AccountId,
+        amount: BalanceOf<T>,
+    ) -> DispatchResult {
+        let execution_id = ExecutionNonce::<T>::get();
+        ExecutionNonce::<T>::mutate(|nonce| *nonce += 1);
+
+        T::Currency::transfer(
+            payee,
+            &T::EscrowAccount::get(),
+            amount,
+            ExistenceRequirement::KeepAlive,
+        )?;
+        T::Currency::reserve(&T::EscrowAccount::get(), amount)?;
+
+        if ExecutionRegistry::<T>::contains_key(execution_id) {
+            return Err(Error::<T>::ExecutionAlreadyRegistered.into())
+        }
+
+        ExecutionRegistry::<T>::insert(
+            execution_id,
+            ExecutionRegistryItem::new(payee.clone(), recipient.clone(), amount),
+        );
+
+        Self::deposit_event(Event::DepositReceived {
+            execution_id,
+            payee: payee.clone(),
+            recipient: recipient.clone(),
+            amount,
+        });
+
+        Ok(())
+    }
+
+    fn finalize(execution_id: ExecutionId, reason: Option<Reason>) -> DispatchResult {
+        let item = Pallet::<T>::execution_registry(execution_id)
+            .ok_or(Error::<T>::ExecutionNotRegistered)?;
+        Self::split(item, reason)?;
+        ExecutionRegistry::<T>::remove(execution_id);
+
+        Self::deposit_event(Event::ExecutionFinalized { execution_id });
+        Ok(())
+    }
+
+    fn issue(recipient: &T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
+        T::Currency::repatriate_reserved(
+            &T::EscrowAccount::get(),
+            recipient,
+            amount,
+            BalanceStatus::Free,
+        )?;
+
+        Self::deposit_event(Event::Issued {
+            recipient: recipient.to_owned(),
+            amount,
+        });
+
+        Ok(())
+    }
+
+    fn split(
+        item: ExecutionRegistryItem<T::AccountId, BalanceOf<T>>,
+        reason: Option<Reason>,
+    ) -> DispatchResult {
+        // Simple rules for splitting, for now
+        let (payee_split, recipient_split): (u8, u8) = match reason {
+            None => (0, 100),
+            Some(Reason::ContractReverted) => (90, 10),
+            Some(Reason::UnexpectedFailure) => (50, 50),
+        };
+
+        let pay_split = |split: u8, recipient: &T::AccountId| -> DispatchResult {
+            if !split.is_zero() {
+                let percent = Percent::from_percent(split);
+                let amt = percent * *item.balance();
+                Self::issue(recipient, amt)
+            } else {
+                Ok(())
+            }
+        };
+
+        // TODO: these need to be joined or handle failure, maybe on_initialize retry a queue of failures after reserving
+        pay_split(payee_split, item.payee())?;
+        pay_split(recipient_split, item.recipient())?;
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Maybe move these tests to the tests.rs file?

Reply:
Ah, the tests.rs is more for integration-like tests. These are unit tests and normally stay in the module